astroquery/mocserver/core.py

```diff
@@ -1,416 +1,707 @@
-
-        if 'meta_data' in kwargs:
-            request_payload.update({'expr': kwargs['meta_data']})
-
-        if 'fields' in kwargs:
-            fields = kwargs['fields']
-            field_l = list(fields) if not isinstance(fields, list) else copy(fields)
-            # The CDS MOC service responds badly to record queries which do not ask
-            # for the ID field. To prevent that, we add it to the list of requested fields
-            field_l.append('ID')
-            field_l = list(set(field_l))
-            fields_str = str(field_l[0])
-            for field in field_l[1:]:
-                fields_str += ', '
-                fields_str += field
-
-            request_payload.update({"fields": fields_str})
-
-        if 'max_rec' in kwargs:
-            max_rec = kwargs['max_rec']
-            request_payload.update({'MAXREC': str(max_rec)})
-
-        self.return_moc = kwargs.get('return_moc', False)
-        if self.return_moc:
-            request_payload.update({'get': 'moc'})
-            if 'max_norder' in kwargs:
-                request_payload.update({'order': kwargs['max_norder']})
-            else:
-                request_payload.update({'order': 'max'})
-
-        return request_payload
-
-    def _parse_result(self, response, *, verbose=False):
+        `~astropy.table.Table`
+            A table containing the field names, heir occurrence (expressed in number
+            of records), and an example for each field.
+
+        Examples
+        --------
+        >>> from astroquery.mocserver import MOCServer
+        >>> MOCServer.list_fields("publisher")  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+        <Table length=3>
+          field_name   occurrence          example
+            str27        int64              str70
+        -------------- ---------- -------------------------
+          publisher_id      32987                 ivo://CDS
+         publisher_did       2871 ivo://nasa.heasarc/warps2
+        hips_publisher         14                       CDS
+
         """
-        Parsing of the response returned by the MOCServer.
+        fields_descriptions = dict(self.list_fields_async(cache=cache).json()[0])
+        occurrences = [
+            int(value.split("x")[0][1:]) for value in fields_descriptions.values()
+        ]
+        field_names = list(fields_descriptions.keys())
+        examples = [value.split("ex: ")[-1] for value in fields_descriptions.values()]
+        list_fields = Table(
+            [field_names, occurrences, examples],
+            names=["field_name", "occurrence", "example"],
+        )
+        list_fields.sort("occurrence", reverse=True)
+        if keyword:
+            return list_fields[
+                [
+                    keyword.casefold() in name.casefold()
+                    for name in list_fields["field_name"]
+                ]
+            ]
+        return list_fields
+
+    def list_fields_async(self, cache=True):
```

Comment:
Should this also have the `keyword` kwarg as the non-async equivalent has is? Do we even need this async?